doc/src/manual/interfaces.md

```diff
@@ -236,6 +236,7 @@ ourselves, we can officially define it as a subtype of an [`AbstractArray`](@ref
 | `axes(A)`                                    | `map(OneTo, size(A))`                  | Return a tuple of `AbstractUnitRange{<:Integer}` of valid indices                    |
 | `similar(A, ::Type{S}, inds)`              | `similar(A, S, Base.to_shape(inds))`   | Return a mutable array with the specified indices `inds` (see below)                  |
 | `similar(T::Union{Type,Function}, inds)`   | `T(Base.to_shape(inds))`               | Return an array similar to `T` with the specified indices `inds` (see below)          |
+| `describe_valid_indices(io, A, i)` |  | Human readable version of the sentence "Valid indices are ...", for the `BoundsError` message |
```

Comment:
For `AbstractArrays` a default fallback which shows `axes` should be enough, no? Feels pretty strange that with the basic methods above you get everything from sorting to matrix multiplication working, but you have to define this extremely specific thing.

Reply:
There is such a fallback, this is in the "optional" section.